### PR TITLE
Rhel9 support on aws

### PR DESCRIPTION
**Pull request by maximenoel8**
## What does this PR change?

- Update user_data to support rhel9 deployment on AWS
- Add rhel 9 support to controller

### CODE REVIEW

File: backend_modules/aws/host/user_data.yaml [hunk 1]
```diff
@@ -118,8 +119,21 @@ yum_repos:
...
+    enabled: true
+    gpgcheck: false
+    name: epel
+    priority: 98
 
 packages: ["salt-minion"]
+runcmd:
+  # WORKAROUND: rhel9 change ssh security for ssh key
+  - echo "PermitRootLogin yes" >> /etc/ssh/sshd_config
+  - update-crypto-policies --set LEGACY
```
Comment (mcalmer) [inline backend_modules/aws/host/user_data.yaml:134]: I am sceptical to use LEGACY here. Can you say why we needs this?

File: backend_modules/aws/host/user_data.yaml [hunk 2]
```diff
@@ -118,8 +119,21 @@ yum_repos:
     gpgcheck: false
     name: temp_tools_pool_repo
     priority: 98
+  epel:
+    baseurl: http://download-cc-rdu01.fedoraproject.org/pub/epel/9/Everything/x86_64/
+    failovermethod: priority
+    enabled: true
+    gpgcheck: false
+    name: epel
+    priority: 98
```
Comment (srbarrios) [inline backend_modules/aws/host/user_data.yaml:128]: Cloud-init stage has as a only goal prepare the instance, but the repositories we will use after the cloud-init stage must go to the salt state part. If we are not installing anything in this image during cloud-init stage, then... I would say we don't need this repository here.

Comment (maximenoel8) [reply backend_modules/aws/host/user_data.yaml:128]: salt-minion is only available in this repository.

File: backend_modules/aws/host/user_data.yaml (hunk 1 above)
Comment (maximenoel8) [reply backend_modules/aws/host/user_data.yaml:134]: With the depreciation of SHA1 on RHEL9, using aws ssh key with terraform is blocked by RHEL9. Without this command, I fail to connect the minion and run salt.

Comment (mcalmer) [reply backend_modules/aws/host/user_data.yaml:134]: Should we better create a new key? @juliogonzalez ?

Comment (juliogonzalez) [reply backend_modules/aws/host/user_data.yaml:134]: What key are we talking about? The key pair created at AWS to connect to the VMs? Is it SHA1 now?

If so, yes, just create an key and use it.

Comment (maximenoel8) [reply backend_modules/aws/host/user_data.yaml:134]: I switch to use an ed25519 ssh key and it works

**Pull request merged**